### DISCUSSION TITLE
QUploader and iOS Devices

### DISCUSSION THREAD

**Discussion by Dirk-** [quasarframework/quasar#11666] [General - Components / Directives / etc]
QUploader not only allows you to upload files, but also accepts pictures directly from the camera of a mobile device.

Unfortunately, iOS devices label all pictures coming from that source `image.jpg`. That means I can only upload one picture at a time. When I take a second picture, nothing happens (only the first picture is shown), even when I upload the first one by touching the cloud symbol before. This works fine with Android devices since they name the pictures arbitrarily. Is it possible to change the name of the picture somehow when it gets accepted by QUploader and before it is displayed?

Here is a CodePen (Quasar v2.2.1) where I tried to resolve this issue: https://codepen.io/Dirk-/pen/LYjdMRb?editors=1010
You can test this behavior on a PC by dragging a file named `image.jpg` onto QUploader. 

No `onFailed` or `onRejected` event is triggered when I add a second file with the same name. I thought I'd change the name of the file on `@added` or in a `:filter` function in case it is called `image.jpg`, but this does not work.

What am I doing wrong?

**Comment by pdanpdan** [comment]
@rstoenescu Maybe we should include the file size in the check for duplicates (https://github.com/quasarframework/quasar/blob/c079f32d05806546b16dc479cfd102094b83a8aa/ui/src/composables/private/use-file.js#L116)?

In fact we should create a synthetic key and use it instead of key all over.
